src/services/util/mod.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Utils for general operations.
+
+pub use builder::*;
+pub use error::*;
```

Comment:
`parse_error` is kept inside services by design. We don't want a general error parser as we plan to introduce services' own error struct in the future.

Reply:
copy this function from azblob

Reply:
That's OK, but keep in `obs` mod is enough. We can do this refactor in another PR>